File: mobile-engage/src/main/java/com/emarsys/mobileengage/iam/PushToInAppAction.java
package com.emarsys.mobileengage.iam;

import android.app.Activity;

import com.emarsys.core.activity.ActivityLifecycleAction;
import com.emarsys.core.provider.timestamp.TimestampProvider;
import com.emarsys.core.util.Assert;

public class PushToInAppAction implements ActivityLifecycleAction {

    private final TimestampProvider timestampProvider;
    private final OverlayInAppPresenter overlayInAppPresenter;
    private final String campaignId;
    private final String html;
    private final String sid;
    private final String url;

    public PushToInAppAction(OverlayInAppPresenter overlayInAppPresenter, String campaignId, String html, String sid, String url, TimestampProvider timestampProvider) {
        Assert.notNull(overlayInAppPresenter, "InAppPresenter must not be null!");
        Assert.notNull(campaignId, "CampaignId must not be null!");
        Assert.notNull(html, "Html must not be null!");
        Assert.notNull(timestampProvider, "TimestampProvider must not be null!");

        this.overlayInAppPresenter = overlayInAppPresenter;
        this.campaignId = campaignId;
        this.html = html;
        this.sid = sid;
        this.url = url;
        this.timestampProvider = timestampProvider;
    }

    @Override
    public void execute(Activity activity) {
        overlayInAppPresenter.present(campaignId, sid, url, null, timestampProvider.provideTimestamp(), html, null);
    }
}

File: mobile-engage/src/main/java/com/emarsys/mobileengage/iam/PushToInAppAction.kt
package com.emarsys.mobileengage.iam

import android.app.Activity
import com.emarsys.core.activity.ActivityLifecycleAction
import com.emarsys.core.activity.ActivityLifecyclePriorities
import com.emarsys.core.provider.timestamp.TimestampProvider

class PushToInAppAction(
    private val overlayInAppPresenter: OverlayInAppPresenter,
    private val campaignId: String,
    private val html: String,
    private val sid: String?,
    private val url: String?,
    private val timestampProvider: TimestampProvider,
    override val priority: Int = ActivityLifecyclePriorities.PUSH_TO_INAPP_ACTION_PRIORITY,
    override val repeatable: Boolean = false,
    override val triggeringLifecycle: ActivityLifecycleAction.ActivityLifecycle = ActivityLifecycleAction.ActivityLifecycle.RESUME
) : ActivityLifecycleAction {
    override fun execute(activity: Activity?) {
        overlayInAppPresenter.present(
            campaignId, sid, url, null, timestampProvider.provideTimestamp(), html, null
        )
    }
}